test/scalacheck/src/test/scala/strawman/collection/immutable/ImmutableChampHashMapProperties.scala

```diff
@@ -0,0 +1,56 @@
+package strawman.collection.immutable
+
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+import org.scalacheck._
+
+import strawman.collection.Generators._
+
+object ImmutableChampHashMapProperties extends Properties("immutable.ChampHashMap") {
+
+  type K = String
+  type V = String
+  type T = (K, V)
+
+  //  override def overrideParameters(p: org.scalacheck.Test.Parameters) =
+  //    p.withMinSuccessfulTests(1000)
+
+  property("convertToScalaMapAndCheckSize") = forAll { (input: ChampHashMap[K, V]) =>
+    convertToScalaMapAndCheckSize(input)
+  }
+
+  private def convertToScalaMapAndCheckSize(input: ChampHashMap[K, V]) =
+    HashMap.from(input).size == input.size
+
+  property("convertToScalaMapAndCheckHashCode") = forAll { (input: ChampHashMap[K, V]) =>
+    convertToScalaMapAndCheckHashCode(input)
+  }
+
+  private def convertToScalaMapAndCheckHashCode(input: ChampHashMap[K, V]) =
+    HashMap.from(input).hashCode == input.hashCode
+
+  property("convertToJavaMapAndCheckEquality") = forAll { (input: ChampHashMap[K, V]) =>
+    input == HashMap.from(input) && HashMap.from(input) == input
+  }
+
+  property("input.equals(duplicate)") = forAll { (inputMap: ChampHashMap[K, V]) =>
+    val builder = ChampHashMap.newBuilder[K, V]()
+    inputMap.foreach(builder.addOne)
+
+    val duplicateMap = builder.result
+    inputMap == duplicateMap
+  }
+
+  property("checkSizeAfterInsertAll") = forAll { (inputValues: HashMap[K, V]) =>
+    val testMap = ChampHashMap.empty[K, V] ++ inputValues
+    inputValues.size == testMap.size
+  }
+
+  property("containsAfterInsert") = forAll { (inputValues: HashMap[K, V]) =>
+    var constructedMap = ChampHashMap.empty[K, V]
+
+    inputValues.foreach(item => constructedMap += item)
+    inputValues.forall(keyValueTuple => constructedMap.get(keyValueTuple._1).get == keyValueTuple._2)
```

Comment:
Instead of `myOption.get == something` you can write `myOption.contains(something)`:

~~~ scala
inputValues.forall { case (key, value) => constructedMap.get(key).contains(value) }
~~~